api/blueprints/drbCollection.py

```diff
@@ -89,6 +89,50 @@ def collectionCreate(user=None):
 
     return APIUtils.formatOPDS2Object(201, opdsFeed)
 
+@collection.route('/update/<uuid>', methods=['POST'])
+@validateToken
+def collectionUpdate(uuid, user=None):
+    logger.info('Updating current collection')
+
+    collectionData = request.json
+    dataKeys = collectionData.keys()
+
+    if len(set(dataKeys) & set(['title', 'creator', 'description'])) < 3\
+            or len(set(dataKeys) & set(['workUUIDs', 'editionIDs'])) == 0:
+        errMsg = {
+            'message':
+                'title, creator and description fields are required'
+                ', with one of workUUIDs or editionIDs to create a collection'
+        }
+
+        return APIUtils.formatResponseObject(400, 'createCollection', errMsg)
+
+    dbClient = DBClient(current_app.config['DB_CLIENT'])
+    dbClient.createSession()
+
+    #Getting the collection the user wants to replace
+    collection = dbClient.fetchSingleCollection(uuid)
+
+    #Creating the new collection that will replace the old collection
+    updatedCollection = dbClient.createCollection(
```

Comment:
hmmm, do we need to add a new method to update the collection with the given uuid instead of creating a new one?

Reply:
We're doing this for now since the base requirements for the ticket is to allow this endpoint to replace all the data of a current collection with a new collection. If we delete the old collection and create a new one, than the id of the new one will be different in the database instead of the same as the old collection. We want to retain the id of the original collection.

Later on, I plan to implement adding/deleting works and editions from a collection instead of simply replacing the whole collection object.